### ISSUE TITLE
Is every style supposed to be output twice?

### ISSUE THREAD

**Issue by coleh33** [jhildenbiddle/css-vars-ponyfill#120]
I am taking over working on a piece of a project that utilizes this feature, however I am seeing something strange that just doesn't make sense.

For every single element that I have CSS declarations for, they are declared twice, when inspecting them in Chrome developer tools.  This is for every element, and not just for elements using CSS variables.

I attached a screenshot of what I am talking about.  Is this the normal intended functionality?

<img width="322" alt="Screen Shot 2020-04-24 at 6 33 49 AM" src="https://user-images.githubusercontent.com/11298821/80218360-f0b38500-85f5-11ea-98f8-56ce90c42468.png">

**Comment by jhildenbiddle** [comment]
Hi @coleh33 --

It is normal functionality, but now that you mentioned it (and I can't believe I hadn't considered this earlier) it probably should not be.

The ponyfill generates and appends new `<style>` nodes to the DOM, but the original `<style>` and/or `<link>` nodes are left in place for subsequent updates. For example:

**Before Ponyfill:**

```html
<link rel="stylesheet" href="style.css" >
```

**After Ponyfill:**

```html
<link rel="stylesheet" href="style.css" data-cssvars="src" data-cssvars-job="1" data-cssvars-group="1">
<style data-cssvars="out" data-cssvars-job="1" data-cssvars-group="1">
  /* Ponyfill-generated CSS */
</style>
```

This is why you are seeing each rule twice: one is the active style from the ponyfill-generated `<style>` node and the other is the stylesheet that the ponyfill generated CSS *from* which is now being overridden (the `<link>` in the above example).

Here's the interesting thing: leaving the source CSS node in the DOM is necessary for subsequent updates, but leaving its associated stylesheet enabled comes at a cost: 1) it increases the amount of CSS the browser has to parse and apply to the DOM (which is by far the slowest part of the entire process) and 2) it causes every declaration to appear twice (as you've point out).

Here's the good news: there is an easy fix that will significantly increase performance and address declarations appearing twice in the developer tools. The fix is simply to disable the stylesheets on source nodes after the ponyfill-generated `<style>` node(s) have been appended to the DOM. This should be baked into the ponyfill, but for now you can accomplish the same thing as follows:

```js
cssVars({
  onSuccess(cssText, elm, url) {
    elm.sheet.disabled = true;
  }
});
```

The above implementation isn't ideal because ponyfill-generated `<style>` nodes haven't been appended when `onSuccess` is invoked, but it's an quick-and-easy implementation that proves the point. Disabling of source node stylesheets should be done after `onComplete` by the ponyfill.

The downside to this approach is that the ponyfill-generated `<style>` elements do not contain source map information so it will be difficult to located a specific declaration by line number using the dev tools. If disabling of source node stylesheets is baked into the ponyfill, I would provide an option to toggle it on/off to make debugging easier (i.e. enabled by default for performance/production, disabled available for development to see source map information if available).

**Comment by jhildenbiddle** [comment]
Added to the backlog. I'll likely push this out the next few days along with #119.

**Comment by jhildenbiddle** [comment]
@coleh33 --

This has been resolved in 2.3.0.
